test/integration/query-interface.test.js

```diff
@@ -53,7 +54,7 @@ describe(Support.getTestDialectTeaser('QueryInterface'), () => {
       await this.sequelize.query(sql);
       let tableNames = await this.queryInterface.showAllTables();
       await cleanup(this.sequelize);
-      if (tableNames[0] && tableNames[0].tableName) {
+      if (tableNames?.[0]?.tableName) {
```

Comment:
The operator applies to the following property/value.  
In this case:
```javascript
tableNames?.[0]
```
Says "if tableNames has an element at index 0" (e.g., `[ {nestedObject} ]` vs `[ ]`) and then

```javascript
tableNames?.[0]?.tableName
```
says if first element contains a property called `tableName`, use it, otherwise return `undefined`.

Reply:
> `tableNames?.[0]` Is testing that there's an element at index 0 of the array, not if tableNames is undefined or null. 

No I guarantee it's testing whether `tableNames` is nullish :)

![image](https://user-images.githubusercontent.com/1280915/172075037-8d2434a0-5b46-4ebc-9e23-da5eb049300a.png)

`tableNames?.[0]` is equivalent to `tableNames == null ? tableNames : tableNames[0]`